src/content/docs/en/reference/experimental-flags/svg-optimization.mdx

```diff
@@ -0,0 +1,225 @@
+To enable this feature, add the experimental flag in your Astro config:
+
+```js title="astro.config.mjs" ins={4-8}
+import { defineConfig } from "astro/config"
+
+export default defineConfig({
+  experimental: {
+    svg: {
+      optimize: true
+    }
+  }
+})
+```
+
+## Usage
+
+No change to using SVG components is required to take advantage of this feature. With experimental `svg.optimization` enabled, all your SVG component import files will be automatically optimized:
+
+```astro title="src/pages/index.astro"
+---
+import Logo from '../assets/logo.svg';
+---
+
+<Logo />
+```
+
+The SVG will be optimized during the build process, resulting in smaller file sizes in your production build.
+
+If SVGO optimization fails for any reason, Astro will gracefully fall back to using the original, unoptimized SVG content. A warning will be logged to the console, but your build will continue without errors.
+
+Note that this optimization applies to every SVG component import in your project. It is not possible to opt out on a per-component basis.
+
+## Configuration
+
+### `optimize`
+
+**Type:** `boolean`<br />
+**Default:** `true`
+
+Whether to enable SVG optimization using SVGO during build time.
+
+When enabled, your imported SVG files used as components will be optimized for smaller file sizes and better performance while maintaining visual quality.
+
+```js title="astro.config.mjs"
+export default defineConfig({
+  experimental: {
+    svg: {
+      optimize: true
+    }
+  }
+})
+```
+
+### `svgoConfig`
+
+**Type:** `Config` (SVGO configuration object)<br />
+**Default:** `{}`
+
+This allows you to further control Astro's SVG optimization with [SVGO plugins](https://svgo.dev/docs/plugins/).
+
+```js title="astro.config.mjs"
+export default defineConfig({
+  experimental: {
+    svg: {
+      optimize: true,
+      svgoConfig: {
+        plugins: [
+          'preset-default',
+          {
+            name: 'removeViewBox',
+            active: false
+          }
+        ]
+      }
+    }
+  }
+})
+```
+
+#### `plugins`
+
+**Type:** `Array<string | PluginConfig>`<br />
+**Default:** `[]`
+
+An array of [SVGO plugins](https://svgo.dev/docs/plugins/) that will be used to optimize your SVG component imports.
+
+This can include any plugins by ID name, including SVGO's `preset-default` collection of plugins. A plugin can optionally be passed as an object including both its `name` and `active` status, to enable or disable as necessary.
+
+```js title="astro.config.mjs"
+export default defineConfig({
+  experimental: {
+    svg: {
+      optimize: true,
+      svgoConfig: {
+        plugins: [
+          'preset-default',
+          {
+            name: 'removeViewBox',
+            active: false
+          }
+        ]
+      }
+    }
+  }
+})
+```
+
+#### Other configuration options
+
+You can also pass [other SVGO configuration options](https://svgo.dev/docs/configuration/), such as `floatPrecision` and `multipass`, directly to your config object:
```

Comment:
This is exactly what I'd want, but the link is 404'ing. So we'll need to find wherever this full configuration is listed! We prefer not to, but if we have to, we can even link to a file/specific lines in their repo where these options are listed. A docs link is better, but we absolutely need to make sure people can find what these options are, one way or another!